### ISSUE TITLE
[Bug]: A11y tests not working with 'react-resizable-panels' npm package

### ISSUE THREAD

**Issue by lukasstransky** [microsoft/playwright#35440]
### Version

1.51.1

### Steps to reproduce

Steps to reproduce:

1. Extract the attached `.zip` file containing the sample app where the issue occurs from the comment.
2. Open the extracted app
3. Run `npm install`
4. Run `playwright test -c playwright-a11y.config.ts --project=chromium`

### Expected behavior

The a11y Playwright dummy test should execute successfully.

### Actual behavior

The a11y Playwright dummy tests fails due the following exception:

```
Error: page._wrapApiCall: TypeError: createContext is not a function
    at http://localhost:3100/assets/App-ANnYG2MI.js:250043:27
    at http://localhost:3100/assets/App-ANnYG2MI.js:250043:27
    at mount (/Users/.../tryout-a-11-y-issue-app/node_modules/@playwright/experimental-ct-core/lib/mount.js:45:35)
    at /Users/.../tryout-a-11-y-issue-app/a11y/tests/dummy-a11y.spec.tsx:7:11
```

### Environment

```shell
System:
    OS: macOS 15.3.2
    CPU: (14) arm64 Apple M3 Max
    Memory: 2.58 GB / 96.00 GB
  Binaries:
    Node: 20.9.0 - ~/.nvm/versions/node/v20.9.0/bin/node
    Yarn: 1.22.22 - /opt/homebrew/bin/yarn
    npm: 10.1.0 - ~/.nvm/versions/node/v20.9.0/bin/npm
    pnpm: 10.6.2 - ~/.nvm/versions/node/v20.9.0/bin/pnpm
  Languages:
    Bash: 3.2.57 - /bin/bash
  npmPackages:
    @playwright/experimental-ct-core: 1.51.1 => 1.51.1 
    @playwright/experimental-ct-react: 1.51.1 => 1.51.1
```

**Comment by lukasstransky** [comment]
[tryout-a-11-y-issue-app.zip](https://github.com/user-attachments/files/19550654/tryout-a-11-y-issue-app.zip)

**Comment by Skn0tt** [comment]
Hi Lukas! This repro contains a bunch of dependencies that I'm not going to install before I audited them. Could you provide me with minimal repro that does not use as many dependencies?

**Comment by Skn0tt** [comment]
I'm getting a different error message with that repo, and it's still not very minimal. If I had to guess, i'd say the problem is around this usage of `createContext` in `react-resizable-panels`. If you look at their `vendor/react.ts` file, there's a blurb about import problems.

I'm still not sure what exactly is causing the problem though. We need a *minimal* repro for that. Please start from a blank new Playwright project, and add no packages except for react-resizable-panels and reproduce the problem. Once you share that with us, we can investigate if it's a bug on our end or (more likely) in this funky react-resizable-panels react code.

**Comment by lukasstransky** [comment]
> I'm getting a different error message with that repo, and it's still not very minimal. If I had to guess, i'd say the problem is around this usage of `createContext` in `react-resizable-panels`. If you look at their `vendor/react.ts` file, there's a blurb about import problems.
> 
> I'm still not sure what exactly is causing the problem though. We need a _minimal_ repro for that. Please start from a blank new Playwright project, and add no packages except for react-resizable-panels and reproduce the problem. Once you share that with us, we can investigate if it's a bug on our end or (more likely) in this funky react-resizable-panels react code.

Sorry for that, this should work now: 

[tryout-a-11-y-issue-app.zip](https://github.com/user-attachments/files/19562845/tryout-a-11-y-issue-app.zip)

**Comment by Skn0tt** [comment]
Thanks, that was very helpful. I was able to narrow it down to this:

```ts
import * as React from 'react';

// this passes:
React.createContext(null);
// but this fails!
const { createContext } = React;
createContext(null);

const App = () => {
    return (
        <div>
            <h1>Hello, World!</h1>
        </div>
    );
};

export default App;
```

I'm investigating further

**Comment by Skn0tt** [comment]
Enabling treeshake here fixes it:

https://github.com/microsoft/playwright/blob/dbd3b6753cb92fba56b105521e1c3d5857e0b308/packages/playwright-ct-core/src/viteUtils.ts#L132

It looks like there's a bug in Rollup. I'll come up with a repro and file with them.

**Comment by Skn0tt** [comment]
There we go: https://github.com/rollup/rollup/issues/5899

**Comment by Skn0tt** [comment]
the upstream issue hasn't yet received much love - I opened a PR with a fix, let's hope that gets more eyes: https://github.com/rollup/rollup/pull/5908

**Comment by lukasstransky** [comment]
Hey, thanks for opening [the PR.](https://github.com/rollup/rollup/issues/5899) They merged it now and released a new version 4.40 with the fix. Which dependency needs now to be updated, or in which package is the rollup package used which caused the problem?

**Comment by Skn0tt** [comment]
We're using Rollup via Vite. The Vite we currently ship has `rollup: ^4.30.1` in their dependencies, which allows upgrading to `4.40`. So I think this should be fixed for you when you make NPM resolve the dependencies again (e.g. with `npm i @playwright/test@latest`)

**Comment by Skn0tt** [comment]
I verified with the repro that the bug is fixed on a clean install. Closing!
